### PR TITLE
ENH: Record property and doc updates

### PR DESCRIPTION
**Pull request by ZLLentz**
## Description

- Add `Daq.record` as a property to schedule a `Daq.configure(record=True)` call later.
- Fix issue where `record` would be reset to `False` after every `configure` where we forget to set it.

## Motivation and Context

This makes it so we can enable/disable recording without seeing the output of `Daq.configure` in `ipython`. The fix means we can create plans that ignore the `record` parameter and leave it to the user to set up.

### CODE REVIEW

**Comment by teddyrendahl** [comment]
I first imagined it as a `property` with getter and setter but that is just a style choice in the end.
```python
daq.record = True
```
The rest looks rather uncontroversial. Nice work on the docs.

**Comment by ZLLentz** [comment]
I was worried about calling `configure` from a property set. I think that's the line I want to draw with properties, setting one should never start a chain reaction of function calls and daq changes.

**Comment by teddyrendahl** [comment]
Interesting. If you just set `self.config['record'] = True` then you would have to call `configure` again later?

**Comment by ZLLentz** [comment]
`self.config` is a dummy dictionary that tells you what you've most recently set. It doesn't affect the real daq at all.

**Comment by ZLLentz** [comment]
You've managed to give me paranoia about someone modifying this dict. I'm going to change `config` to return a copy.

**Comment by teddyrendahl** [comment]
Right. I'm just suggesting you do:

```python
@property
def record(self):
    return self.config['record']

@record.setter
def record(self, value):
     self.config['record'] = value
     self._configured = False
```
Then next time you call `kickoff` it'll configure and make your `self.config` changes live. Once again, just a style thing. Makes it easier to check if you are recording or not.

**Comment by ZLLentz** [comment]
I see what you mean. I could implement this, but it's not as straightforward as you are suggesting because `kickoff` only calls `configure` if `configure` has never been called. It would also need to inspect the `config` dict and compare it to the last `configure` call.

**Comment by teddyrendahl** [comment]
I see. Think you would just need to change the `configured` property to keep track of whether `config` has been pushed to the DAQ or not. Slightly more complex then just checking if there is a `_config` dictionary at all.

Up to you. I've voiced my opinion. I am not sure how often users are going to want to use the Python to check if the DAQ is recording or not.

**Comment by ZLLentz** [comment]
The thing is that, semantically, there is no "unconfiguring" the daq from python. One `configured` is `True` it will always be `True` until we `disconnect`. Changing the property will make the class confusing because we will have `status` returning `'Configured'` and `configured` returning `False`.

I can implement `record` as a property but this isn't the way to do it. There currently is no support in the class for setting a config now and configuring later, but this might be desirable in general.

**Comment by teddyrendahl** [comment]
Yeah I see. There is merit in making sure you leave the `config` alone as a pure readback of what you have configured the `DAQ` object. Don't think it would be too hard to have a `_reconfigure` flag that you keep internally.

The reason I am sensitive to this `record` property in particular is I think it is the one choice that we shouldn't be making at the plan level. I'm guessing most of the other configuration will be done by the `daq_wrapper` but I see the following workflow:

1. Write plan with daq_wrapper setting DAQ mode, events ...
2. Try it a few times to see it works with DAQ running but not recording. Fix bugs.
3. Run it for reals with record.

Ideally between steps 2 and 3 you just flick that `DAQ.record` and do not change the code in your plan. 

In conclusion, do what you feel. I like the property but I like properties in general. Gives you both readback and setting abilities 🤷‍♂️

**Comment by ZLLentz** [comment]
We talked a bit offline if anyone is following this. I don't dislike properties, I just wanted to emphasize that this isn't a tiny fix. I'm going to take the time to implement `record` as a property and to enable this and other methods to store a future configuration to be configured later.

**Comment by ZLLentz** [comment]
I spent more time fighting doing work here than I spent implementing this as a property to be configured later. My b
